### ISSUE TITLE
Exporting Tasks

### ISSUE THREAD

**Issue by xnacly** [xNaCly/opusCli#6]
Enable exporting to:

- [ ] csv
- [ ] json

**Comment by AntoniosBarotsis** [comment]
Hello! I was looking for Rust issues for [Hacktober](https://hacktoberfest.com/participation/#maintainers) and found this. I see that you aren't explicitly participating but regardless I wouldn't mind helping with this; I'm relatively new to Rust and this seems interesting learning-wise.

I could try working on this in the next few days if you don't mind :)

**Comment by xnacly** [comment]
Sure, have at it, good luck. 

Id suggest you take a look at the sqlite docs, maybe there is existing tooling for this usecase.

**Comment by AntoniosBarotsis** [comment]
Cool, thanks!

Could you add a very basic "getting started" example in your README at some point? Not sure if you need something external (for instance SQLite) to run this.

**Comment by xnacly** [comment]
Sure, i'll do that

**Comment by xnacly** [comment]
> Cool, thanks!
> 
> Could you add a very basic "getting started" example in your README at some point? Not sure if you need something external (for instance SQLite) to run this.

I added a guide, happy hacking :)

**Comment by AntoniosBarotsis** [comment]
I messed around with this for a bit and it *seems* that there is not a built-in way of exporting the data, see [here](https://github.com/rusqlite/rusqlite/discussions/1210).

TLDR: The exports work through the CLI and not through the C library; thus, the Rust wrapper can't do that either. I'll probably make this work with Serde tomorrow.

**Comment by xnacly** [comment]
> I messed around with this for a bit and it _seems_ that there is not a built-in way of exporting the data, see [here](https://github.com/rusqlite/rusqlite/discussions/1210).
> 
> TLDR: The exports work through the CLI and not through the C library; thus, the Rust wrapper can't do that either. I'll probably make this work with Serde tomorrow.

I dont know Serde, but it looks alright

**Comment by AntoniosBarotsis** [comment]
Ok it was actually easier than I thought but I'll leave the PR for tomorrow so I have a clear mind 😄 

![image](https://user-images.githubusercontent.com/50240570/194175807-19218ca2-64ad-47b3-bc3b-691360151ecb.png)
![image](https://user-images.githubusercontent.com/50240570/194175818-cd8a8cf2-361a-4185-a8e9-97c4f54ca1b6.png)

**Comment by xnacly** [comment]
> Ok it was actually easier than I thought but I'll leave the PR for tomorrow so I have a clear mind smile
> 
> ![image](https://user-images.githubusercontent.com/50240570/194175807-19218ca2-64ad-47b3-bc3b-691360151ecb.png) ![image](https://user-images.githubusercontent.com/50240570/194175818-cd8a8cf2-361a-4185-a8e9-97c4f54ca1b6.png)

Looks awesome, im looking forward to your pr
